subsys/shell/shell_uart.c

```diff
@@ -27,62 +27,68 @@ SHELL_DEFINE(shell_uart, CONFIG_SHELL_PROMPT_UART, &shell_transport_uart,
 	     SHELL_FLAG_OLF_CRLF);
 
 #ifdef CONFIG_SHELL_BACKEND_SERIAL_INTERRUPT_DRIVEN
+static inline bool smp_rx_byte(const struct shell_uart *sh_uart, uint8_t byte)
+{
+#ifdef CONFIG_MCUMGR_SMP_SHELL
+	return smp_shell_rx_byte(&sh_uart->ctrl_blk->smp, byte);
+#else
+	return false;
+#endif
+}
+
 static void uart_rx_handle(struct device *dev,
 			   const struct shell_uart *sh_uart)
 {
 	uint8_t *data;
```

Comment:
Lets comment meaning of these variables. Every new entity that wanders into this function needs to figure this out again by itself.